src/hooks/useLiveLogs.ts

```
@@ -0,0 +1,408 @@
+      }
+    } catch (error) {
+      console.error("Failed to parse log message", error);
+    }
+  }, [logs, connected]);
+  
+  // Standard log handler for regular (non-aggregated) logs
+  const addStandardLog = (activityType: string, data: any) => {
+    const logEntry: LogEntry = {
+      id: `log-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`,
+      type: activityType,
+      message: generateLogMessage(activityType, data),
```

ESLint

[error] 'generateLogMessage' was used before it was defined.

(@typescript-eslint/no-use-before-define)